evm/p2p/peer.py

```diff
@@ -710,8 +722,8 @@ def __init__(self,
             if peer is not None:
                 self.logger.info("Successfully connected to %s", peer)
                 self.connected_nodes[peer.remote] = peer
-                asyncio.ensure_future(peer.run(finished_callback=self._peer_finished))
-                self.new_peer_callback(peer)
+                for subscriber in self._subscribers:
+                    subscriber.register_peer(peer)
```

Comment:
Any thoughts on allowing subscribers to unsubscribe themselves by doing something like raising a certain exception.  No clue if it's even useful, but it did jump out to me that there is no mechanism for removing a subscription.

Reply:
Yeah, that will be necessary as we'll have services that will be short-lived (e.g the `StateDownloader`, which I just realized has been broken by those changes), but I believe it'd be better for them to explicitly unsubscribe once they're finished.

I'll try to come up with a simple test for `StateDownloader`, fix it and see what an unsubscribe API would look like.